tests/backup/backup_test.go

```diff
@@ -4679,3 +4824,142 @@ func createUsers(numberOfUsers int) []string {
+func DeleteCloudAccounts(backupLocationMap map[string]string, credName string, cloudCredUID string, ctx context.Context) {
+	if len(backupLocationMap) != 0 {
+		for backupLocationUID, bkpLocationName := range backupLocationMap {
+			DeleteBackupLocation(bkpLocationName, backupLocationUID, orgID)
+		}
+		time.Sleep(time.Minute * 3)
+		DeleteCloudCredential(credName, orgID, cloudCredUID)
+	}
+	DeleteCluster(SourceClusterName, orgID, ctx)
+	DeleteCluster(destinationClusterName, orgID, ctx)
+}
+
+// AddRoleAndAccessToUsers assigns role and access level to the users
+// AddRoleAndAccessToUsers return map whose key is userRoleAccess and value is backup for each user
+func AddRoleAndAccessToUsers(users []string, backupNames []string) (map[userRoleAccess]string, error) {
+	var access BackupAccess
+	var role backup.PxBackupRole
+	RoleAccessUserBackupContext := make(map[userRoleAccess]string)
+	ctx, err := backup.GetAdminCtxFromSecret()
+	log.FailOnError(err, "Fetching px-central-admin ctx")
+	for i := 0; i < len(users); i++ {
+		userIndex := i % 9
+		switch userIndex {
+		case 0:
+			access = ViewOnlyAccess
+			role = backup.ApplicationOwner
+		case 1:
+			access = RestoreAccess
+			role = backup.ApplicationOwner
+		case 2:
+			access = FullAccess
+			role = backup.ApplicationOwner
+		case 3:
+			access = ViewOnlyAccess
+			role = backup.ApplicationUser
+		case 4:
+			access = RestoreAccess
+			role = backup.ApplicationUser
+		case 5:
+			access = FullAccess
+			role = backup.ApplicationUser
+		case 6:
+			access = ViewOnlyAccess
+			role = backup.InfrastructureOwner
+		case 7:
+			access = RestoreAccess
+			role = backup.InfrastructureOwner
+		case 8:
+			access = FullAccess
+			role = backup.InfrastructureOwner
+		default:
+			access = ViewOnlyAccess
+			role = backup.ApplicationOwner
+		}
+		ctxNonAdmin, err := backup.GetNonAdminCtx(users[i], "Password1")
+		log.FailOnError(err, "Fetching user ctx")
+		userRoleAccessContext := userRoleAccess{users[i], role, access, ctxNonAdmin}
+		RoleAccessUserBackupContext[userRoleAccessContext] = backupNames[i]
+		err = backup.AddRoleToUser(users[i], role, "Adding role to user")
+		if err != nil {
+			err = fmt.Errorf("Failed to add role %s to user  %s with err %v", role, users[i], err)
+			return nil, err
+		}
+		err = ShareBackup(backupNames[i], nil, []string{users[i]}, access, ctx)
+		if err != nil {
+			return nil, err
+		}
+		log.Infof(" Backup %s shared with user %s", backupNames[i], users[i])
+	}
+	return RoleAccessUserBackupContext, nil
+}
+
+func ValidateSharedBackupWithUsers(key userRoleAccess, backupName string, restoreName string) {
+	ctx, err := backup.GetAdminCtxFromSecret()
+	log.FailOnError(err, "Fetching px-central-admin ctx")
+	log.InfoD("Registering Source and Destination clusters from user context")
+	CreateSourceAndDestClusters(orgID, "", "", key.context)
+	log.InfoD("Validating if user [%s] with access [%v] and [%v] role can restore and delete backup or not", key.user, key.accesses, key.roles)
```

Comment:
See if you can change the access to a string instead of printing out the int value